src/java/org/apache/cassandra/cql3/statements/schema/CopyTableStatement.java

```diff
@@ -0,0 +1,175 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.statements.schema;
+
+import org.apache.cassandra.audit.AuditLogContext;
+import org.apache.cassandra.audit.AuditLogEntryType;
+import org.apache.cassandra.auth.Permission;
+import org.apache.cassandra.cql3.CQLStatement;
+import org.apache.cassandra.cql3.QualifiedName;
+import org.apache.cassandra.db.guardrails.Guardrails;
+import org.apache.cassandra.exceptions.AlreadyExistsException;
+import org.apache.cassandra.schema.Indexes;
+import org.apache.cassandra.schema.KeyspaceMetadata;
+import org.apache.cassandra.schema.Keyspaces;
+import org.apache.cassandra.schema.TableId;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.schema.Triggers;
+import org.apache.cassandra.schema.UserFunctions;
+import org.apache.cassandra.service.ClientState;
+import org.apache.cassandra.service.reads.repair.ReadRepairStrategy;
+import org.apache.cassandra.tcm.ClusterMetadata;
+import org.apache.cassandra.transport.Event.SchemaChange;
+
+public final class CopyTableStatement extends AlterSchemaStatement
+{
+    private final String sourceKeyspace;
+    private final String sourceTableName;
+    private final String targetKeyspace;
+    private final String targetTableName;
+
+    public CopyTableStatement(String sourceKeyspace,
+                              String targetKeyspace,
+                              String sourceTableName,
+                              String targetTableName)
+    {
+        super(targetKeyspace);
+        this.sourceKeyspace = sourceKeyspace;
+        this.targetKeyspace = targetKeyspace;
+        this.sourceTableName = sourceTableName;
+        this.targetTableName = targetTableName;
+    }
+
+    @Override
+    SchemaChange schemaChangeEvent(Keyspaces.KeyspacesDiff diff)
+    {
+        return new SchemaChange(SchemaChange.Change.CREATED, SchemaChange.Target.TABLE, targetKeyspace, targetTableName);
+    }
+
+    @Override
+    public void authorize(ClientState client)
+    {
+        client.ensureTablePermission(sourceKeyspace, sourceTableName, Permission.SELECT);
+        client.ensureAllTablesPermission(targetKeyspace, Permission.CREATE);
+    }
+
+    @Override
+    public AuditLogContext getAuditLogContext()
+    {
+        return new AuditLogContext(AuditLogEntryType.CREATE_TABLE_LIKE, targetKeyspace, targetTableName);
+    }
+
+    public String toString()
+    {
+        return String.format("CREATE TABLE %s.%s LIKE %s.%s", targetKeyspace, targetTableName, sourceKeyspace, sourceTableName);
+    }
+
+    @Override
+    public void validate(ClientState state)
```

Comment:
why do we need this? Will not be `validate` of super called automatically when you do not override?

Reply:
You also do not need to do this?

https://github.com/apache/cassandra/pull/3721/files#r1871469974